public/lesswrong-editor/src/ckeditor5-cta-button/cta-button-form.js

```diff
@@ -0,0 +1,363 @@
+      },
+    });
+  }
+
+  get buttonText() {
+    return this.buttonTextView.element.value;
+  }
+
+  set buttonText(value) {
+    this.buttonTextView.element.value = value;
+  }
+
+  get linkTo() {
+    return this.linkToView.element.value;
+  }
+
+  set linkTo(value) {
+    this.linkToView.element.value = value;
+  }
+
+  _createInputs() {
+    // Create 'Button text' input
+    const buttonTextView = new InputTextView(this.locale);
+    const buttonTextBind = buttonTextView.bindTemplate;
+    buttonTextView.setTemplate({
+      tag: "input",
+      attributes: {
+        type: "text",
+        class: ["ck-cta-form-input", buttonTextBind.if("hasError", "ck-error")],
+        id: buttonTextBind.to("id"),
+        placeholder: buttonTextBind.to("placeholder"),
+        readonly: false,
+        "aria-invalid": buttonTextBind.if("hasError", true),
+        "aria-describedby": buttonTextBind.to("ariaDescribedById"),
+      },
+      on: {
+        input: buttonTextBind.to("input"),
+      },
+    });
+    buttonTextView.label = "Button text";
+    buttonTextView.on("input", () => {
+      const model = this.editor.model;
+      const selectedElement = this.selectedElement;
+
+      model.change((writer) => {
+        if (!selectedElement) return;
+
+        const inputValue = buttonTextView.element.value.trim();
+
+        // Remove the existing text nodes within the selected element
+        const range = writer.createRangeIn(selectedElement);
+        writer.remove(range);
+
+        // Insert the new text node with the input value
+        const newText = writer.createText(inputValue);
+        writer.insert(newText, writer.createPositionAt(selectedElement, 0));
+      });
+    });
+
+    // Create 'Link to' input
+    const linkToView = new InputTextView(this.locale);
+    const linkToBind = linkToView.bindTemplate;
+    linkToView.setTemplate({
+      tag: "input",
+      attributes: {
+        type: "text",
+        class: ["ck-cta-form-input", linkToBind.if("hasError", "ck-error")],
+        id: linkToBind.to("id"),
+        placeholder: 'https://example.com',
+        readonly: false,
+        "aria-invalid": linkToBind.if("hasError", true),
+        "aria-describedby": linkToBind.to("ariaDescribedById"),
+      },
+      on: {
+        input: linkToBind.to("input"),
+      },
+    });
+    linkToView.label = "Link to";
+    linkToView.on("input", () => {
+      const model = this.editor.model;
+      const selectedElement = this.selectedElement;
+
+      model.change((writer) => {
+        if (!selectedElement) return;
+
+        const inputValue = linkToView.element.value.trim();
+
+        // Set the 'href' attribute to the new inputValue
+        writer.setAttribute("href", inputValue, selectedElement);
+      });
+    });
+
+    return {
+      buttonTextView,
+      linkToView,
+    };
+  }
+}
+
+/**
+ * Plugin for the form to edit the text and link of a CTA button
+ */
+export default class CTAButtonForm extends Plugin {
```

Comment:
I think we definitely need some validation of the URL somewhere (maybe we can borrow from `url-validator-plugin.js`?)

I was able to create a button with the url `javascript:alert(1)` which is pretty scary - it doesn't actually get executed because of the `target="_blank"`, but it still seems risky and potential escapable.